src/coreclr/src/gc/gc.cpp

```diff
@@ -22032,7 +22311,40 @@ void gc_heap::plan_phase (int condemned_gen_number)
         )
     {
 #ifndef MULTIPLE_HEAPS
-        _sort (&mark_list[0], mark_list_index-1, 0);
+#ifdef USE_VXSORT
+        ptrdiff_t entry_count = mark_list_index - mark_list;
+        // conservatively use AVX2 only for large mark lists,
+        // and do runtime test to check whether AVX2 is indeed available 
+        if (entry_count > AVX2_THRESHOLD_SIZE && IsSupportedInstructionSet (InstructionSet::AVX2))
+        {
+            int32_t* mark_list_32 = (int32_t*)mark_list;
+            uint8_t* low = slow;
+            ptrdiff_t range = shigh - low;
+            if ((uint32_t)range == range)
+            {
+                do_pack_avx2 (mark_list, entry_count, low);
+                _sort (&mark_list_32[0], &mark_list_32[entry_count - 1], 0);
+                do_unpack_avx2 (mark_list_32, entry_count, low);
+#ifdef _DEBUG
+                uint8_t*high = heap_segment_allocated (ephemeral_heap_segment);
+                for (ptrdiff_t i = 0; i < entry_count; i++)
+                {
+                    uint8_t* item = mark_list[i];
+                    assert (low <= item && item < high);
```

Comment:
you could changed this assert to

`assert ((low <= item) && (item <= shigh));`